lambdas/thumbnail/index.py

```diff
@@ -0,0 +1,92 @@
+"""
+Generate thumbnails for images in S3.
+"""
+import base64
+from io import BytesIO
+import json
+import os
+
+from PIL import Image
+import requests
+
+from t4_lambda_shared.decorator import api, validate
+
+
+# Eventually we'll want to precompute/cache thumbnails, so we won't be able to support
+# arbitrary sizes. Might as well copy Dropbox' API:
+# https://www.dropbox.com/developers/documentation/http/documentation#files-get_thumbnail
+SUPPORTED_SIZES = [
+    (32, 32),
+    (64, 64),
+    (128, 128),
+    (256, 256),
+    (480, 320),
+    (640, 480),
+    (960, 640),
+    (1024, 768),
+    (2048, 1536)
+]
+# Map URL parameters to actual sizes, e.g. 'w128h128' -> (128, 128)
+# TODO(dima): Is this a good idea? Don't know why Dropbox does it that way.
```

Comment:
alternatives require parsing strings from the request, and that's strictly worse in performance and readability. it's kinda weird that `f'W{w}H{h}` is not supported but, meh, ppl need call the API right.

Reply:
you could `.lower()` the string before looking it up

Reply:
Ok. Though for whatever map we pick, I think being strict (i.e., case-sensitive) is better. The other URL parameters we use - "ipynb" in the preview lambda, etc. - are case sensitive.

Reply:
yep agreed